core/security_group_rule.go

```diff
@@ -0,0 +1,191 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/krystal/go-katapult"
+)
+
+type SecurityGroupRule struct {
+	ID        string   `json:"id,omitempty"`
+	Direction string   `json:"direction,omitempty"`
+	Protocol  string   `json:"protocol,omitempty"`
+	Ports     string   `json:"ports,omitempty"`
+	Targets   []string `json:"targets,omitempty"`
+	Notes     string   `json:"notes,omitempty"`
+}
+
+func (sgr *SecurityGroupRule) Ref() SecurityGroupRuleRef {
+	return SecurityGroupRuleRef{ID: sgr.ID}
+}
+
+type SecurityGroupRuleRef struct {
+	ID string `json:"id,omitempty"`
+}
```

Comment:
There's no JSON marshaling tests for this struct, though I've noticed that's actually an oversight missing from nearly every single `*Ref` type we have, and needs to be addressed. But let's add it for security group rules here to begin with :)